core/lib/types/src/kzg.rs

```diff
@@ -0,0 +1,315 @@
+use std::convert::TryInto;
+
+use c_kzg::{Blob, Bytes32, KzgCommitment, KzgProof, KzgSettings, BYTES_PER_BLOB};
+use zk_evm::sha3::{Digest, Keccak256};
+use zkevm_circuits::eip_4844::{
+    input::{BLOB_CHUNK_SIZE, ELEMENTS_PER_4844_BLOCK},
+    zksync_pubdata_into_ethereum_4844_data,
+};
+
+const BYTES_PER_BLOB_ZK_SYNC: usize = BLOB_CHUNK_SIZE * ELEMENTS_PER_4844_BLOCK;
+
+/// Packed pubdata commitments.
+/// Format: opening point (16 bytes) || claimed value (32 bytes) || commitment (48 bytes) || opening proof (48 bytes)) = 144 bytes
+const BYTES_PER_PUBDATA_COMMITMENT: usize = 144;
+
+pub struct KzgInfo {
+    pub blob: Blob,
+    pub kzg_commitment: KzgCommitment,
+    pub opening_point: Bytes32,
+    pub opening_value: Bytes32,
+    pub opening_proof: KzgProof,
+    pub versioned_hash: Bytes32,
+    pub blob_proof: KzgProof,
+}
+
+impl KzgInfo {
+    /// Size of KzgInfo is equal to size(blob) + size(kzg_commitment) + size(bytes32) + size(bytes32) + size(kzg_proof) + size(bytes32) + size(kzg_proof)
+    const SERIALIZED_SIZE: usize = 131_072 + 48 + 32 + 32 + 48 + 32 + 48;
+
+    /// Load in local kzg settings file at trusted_setup.txt
+    pub fn kzg_settings() -> KzgSettings {
+        let zksync_home = std::env::var("ZKSYNC_HOME").unwrap_or_else(|_| ".".into());
+        let path = std::path::Path::new(&zksync_home).join("trusted_setup.txt");
+        KzgSettings::load_trusted_setup_file(&path).unwrap()
+    }
```

Comment:
Could you move it to config? 
usng env outside config crate is strictly prohibited